sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2001,8 +2001,10 @@ object SQLConf {
     buildConf("spark.sql.streaming.stateStore.compression.codec")
       .internal()
       .doc("The codec used to compress delta and snapshot files generated by StateStore. " +
-        "By default, Spark provides four codecs: lz4, lzf, snappy, and zstd. You can also " +
-        "use fully qualified class names to specify the codec. Default codec is lz4.")
+        "It is also applied to RocksDB State Store's RocksDB compression type if possible. By " +
+        "default, Spark provides four codecs: lz4, lzf, snappy, and zstd. You can also " +
```

Comment:
> ... Spark provides four codecs: lz4, lzf, snappy, and zstd.

seems rocksdb does not support lzf.

if decide to reuse this conf, description needs to udpate

Reply:
Personally, I wouldn't recommend reusing this conf.

Reply:
If the supported formats do not exactly match, then probably yes, we may need another config specifically for RocksDB.
@siying WDYT?

Reply:
@HeartSaVioR I can do that if that's what you recommend.

Reply:
Yeah... let's do that. I understand we have too many knobs, but it would be harder to explain if we go with one config for two configs which are not 100% compatible. Also the available codecs are tied with RocksDB and we have to track the change and reflect to the doc. That doesn't sound to me to be trivial - we just want to delegate it to RocksDB doc, like we do for Kafka for a lot of options.